### PR TITLE
rsz: preserve buffers on hierarchical module boundaries in canRemoveBuffer

### PR DESCRIPTION
**Pull request by oharboe**
### Summary
When global placement performs timing-driven buffer removal (`Resizer::removeBuffers`), `canRemoveBuffer` checks whether a candidate buffer can be safely removed. If the buffer is situated on a hierarchical module boundary where its input pin and output pin belong to different `dbModNet`s, removing the buffer flattens the net without updating hierarchical module port/net bindings, leaving downstream module inputs disconnected.

### Fix
Check whether `input_modnet != output_modnet` in `Resizer::canRemoveBuffer` and return `false` to preserve buffers located on hierarchical module boundaries.

### CODE REVIEW

File: src/rsz/src/Resizer.cc [hunk 1]
```diff
@@ -2782,6 +2782,11 @@ bool Resizer::canRemoveBuffer(sta::Instance* buffer,
   sta::Pin* output_pin = db_network_->findPin(buffer, output_port);
   sta::Net* input_net = db_network_->net(input_pin);
   sta::Net* output_net = db_network_->net(output_pin);
+  odb::dbModNet* input_modnet = db_network_->hierNet(input_pin);
+  odb::dbModNet* output_modnet = db_network_->hierNet(output_pin);
+  if (input_modnet != output_modnet) {
+    return false;
+  }
```
[the remark this answers is not in the document]
Comment (oharboe) [reply src/rsz/src/Resizer.cc:2789]: Updated — added the `survivor_modnet != removed_modnet` check in `Resizer::removeBuffer` before calling `mergeModNet` to prevent self-merging and destroying the dbModNet.

**Comment by oharboe** [comment]
@precisionmoon Heads up! I changed reviewer because reviewer was automatically assigned and this is a continuation of a conversation between me and @jhkim-pii

**Comment by oharboe** [comment]
@jhkim-pii I don't know what I'm doing here... I don't know if I should follow Gemini's directions... Please advice.

**Comment by gadfort** [comment]
@jhkim-pii isn't the issue here that the hier net is not correctly preserved. The resizer should be allowed to remove buffers on boundaries, but obviously if the net is getting flattened, then there is an issue with the net handling

**Comment by oharboe** [comment]
@jhkim-pii Perhaps I should close this issue and the confidential issue I filed is more helpful? I don't know what I'm doing here, but I have a patch hack so I'm unstuck.

**Comment by oharboe** [comment]
@jhkim-pii I'm closing this PR, please re-open it if it is helpful. I think it is faster for you to examine my reproduction issue than to look at my failed attempts of fixing it.

**Comment by jhkim-pii** [comment]
@oharboe @gadfort Yes. The fix in this PR looks weird to me. Maybe, @oharboe 's confidential test case suffers from a corner-case issue. Debugging the test case and finding the root-cause is the best.

**Comment by jhkim-pii** [comment]
@oharboe @gadfort I found a bug of `remove_buffer` regarding feed-through and fixed it in this PR https://github.com/The-OpenROAD-Project/OpenROAD/pull/11001

**Comment by oharboe** [comment]
> @oharboe @gadfort I found a bug of `remove_buffer` regarding feed-through and fixed it in this PR #11001

🥳
